### ISSUE TITLE
Specs2 Test Interface 'afterAll' not calling super method

### ISSUE THREAD

**Issue by gmalouf** [akka/akka-http#1450]
https://github.com/akka/akka-http/blob/master/akka-http-testkit/src/main/scala/akka/http/scaladsl/testkit/TestFrameworkInterface.scala

The ScalaTest impl calls their afterAll, but the Specs2 one does not.  This actually causes some pain in testing frameworks as we have to override the afterall and call the 'cleanup' directly.

**Comment by jrudolph** [comment]
How would that work exactly? `afterAll` is abstract in specs2's `AfterAll` so it would only matter with more complex `trait` inheritance chains. Is that what happens in your case?

**Comment by gmalouf** [comment]
Good point.  Our use case is with docker-it-scala which also defintes
before/after all methods that do not call a super.
https://github.com/whisklabs/docker-it-scala/blob/master/specs2/src/main/scala/com/whisk/docker/specs2/DockerTestKit.scala

We were fortunate that the methods docker-it-scala is calling in their
before/after alls are public as are the ones in route testkit such that I
could implement my own before/after all that calls everything to start/stop
the db and the testkit..

**Comment by jrudolph** [comment]
I see, I guess it's good practice to call super for that method anyways.

**Comment by gmalouf** [comment]
I think so - not unusual to mix different test frameworks together (i.e.
docker containers and akka test-kit) to build an integration test suite.

**Comment by jrudolph** [comment]
Yeah, just make sure that really all testkits then call the super method (as you did here). Otherwise, it depends on the order of `extends` declarations which will actually be executed.

**Comment by jrudolph** [comment]
PR would be welcome :)

**Comment by gmalouf** [comment]
Will do.

**Comment by gmalouf** [comment]
@jrudolph without adding 'abstract override', we get the following error:
`TestFrameworkInterface.scala:42: method afterAll in trait AfterAll is accessed from super. It may not be abstract unless it is overridden by a member declared `abstract' and `override`

However, adding abstract override forces the end user to mix-in another trait that implements afterAll directly without providing an override.

The ideal solution to this imo is for specs2 to do what ScalaTest does and provide a default implementation; i.e. `def afterAll() = ()` or something  similar.

In the meantime, users that mix multiple frameworks that have their own afterAll() impls have to override and provide their own concrete implementations that duplicate the behavior.
